cmd/tempo/app/modules.go

```diff
@@ -185,6 +185,8 @@ func (t *App) initQueryFrontend() (services.Service, error) {
 	// http query endpoint
 	t.server.HTTP.Handle(queryEndpoint(&t.cfg), tracesHandler)
 
+	t.server.HTTP.Handle("/api/echo", echoHandler())
```

Comment:
this should use `queryEndpoint` like the endpoint above it. it will need to support the prefix so we can read it through our gateway

Reply:
Oh right, good catch!